pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -214,276 +210,271 @@ let
+        ${optionalString (enableSharedExecutables && isExecutable && !isGhcjs && stdenv.isDarwin && stdenv.lib.versionOlder ghc.version "7.10") ''
+          for exe in "$out/bin/"* ; do
+            install_name_tool -add_rpath "$out/lib/ghc-${ghc.version}/${pname}-${version}" "$exe"
+          done
+        ''}
+
+        ${optionalString enableSeparateDocOutput ''
+        for x in ${docdir "$doc"}"/html/src/"*.html; do
+          remove-references-to -t $out $x
+        done
+        mkdir -p $doc
+        ''}
+        ${optionalString enableSeparateDataOutput "mkdir -p $data"}
+
+        runHook postInstall
       '';
-    };
-  };
 
-  meta = { inherit homepage license platforms; }
-         // optionalAttrs broken               { inherit broken; }
-         // optionalAttrs (description != "")  { inherit description; }
-         // optionalAttrs (maintainers != [])  { inherit maintainers; }
-         // optionalAttrs (hydraPlatforms != null) { inherit hydraPlatforms; }
-         ;
-
-}
-// optionalAttrs (preCompileBuildDriver != "")  { inherit preCompileBuildDriver; }
-// optionalAttrs (postCompileBuildDriver != "") { inherit postCompileBuildDriver; }
-// optionalAttrs (preUnpack != "")      { inherit preUnpack; }
-// optionalAttrs (postUnpack != "")     { inherit postUnpack; }
-// optionalAttrs (patches != [])        { inherit patches; }
-// optionalAttrs (patchPhase != "")     { inherit patchPhase; }
-// optionalAttrs (preConfigure != "")   { inherit preConfigure; }
-// optionalAttrs (postConfigure != "")  { inherit postConfigure; }
-// optionalAttrs (preBuild != "")       { inherit preBuild; }
-// optionalAttrs (postBuild != "")      { inherit postBuild; }
-// optionalAttrs (doBenchmark)          { inherit doBenchmark; }
-// optionalAttrs (checkPhase != "")     { inherit checkPhase; }
-// optionalAttrs (preCheck != "")       { inherit preCheck; }
-// optionalAttrs (postCheck != "")      { inherit postCheck; }
-// optionalAttrs (preInstall != "")     { inherit preInstall; }
-// optionalAttrs (installPhase != "")   { inherit installPhase; }
-// optionalAttrs (postInstall != "")    { inherit postInstall; }
-// optionalAttrs (preFixup != "")       { inherit preFixup; }
-// optionalAttrs (postFixup != "")      { inherit postFixup; }
-// optionalAttrs (dontStrip)            { inherit dontStrip; }
-// optionalAttrs (hardeningDisable != []) { inherit hardeningDisable; }
-// optionalAttrs (stdenv.buildPlatform.libc == "glibc"){ LOCALE_ARCHIVE = "${glibcLocales}/lib/locale/locale-archive"; }
-)
+      passthru = passthru // {
+
+        inherit pname version;
+
+        compiler = ghc;
+
+
+        getBuildInputs = {
+          inherit propagatedBuildInputs otherBuildInputs allPkgconfigDepends;
+          haskellBuildInputs = isHaskellPartition.right;
+          systemBuildInputs = isHaskellPartition.wrong;
+        };
+
+        env = buildHaskellPackages.shellFor {
+          packages = p: [ drv ];
```

Comment:
I don't think `buildHaskellPackages` is right.

Reply:
@Infinisil Does it control where GHC comes from? This will get the GHC for targeting the build platform.

Reply:
Ah you're right, will have to do something about this then